### PR TITLE
Feature/m2 1168 add confirm password flow on applet creation edition

### PR DESCRIPTION
**Pull request by alex-i-e**

### CODE REVIEW

File: src/modules/Dashboard/features/Applet/Password/EnterAppletPassword/EnterAppletPassword.tsx [hunk 1]
```diff
@@ -29,6 +29,12 @@ export const EnterAppletPassword = forwardRef<AppletPasswordRef, EnterAppletPass
     const [showPassword, setShowPassword] = useState(false);
 
     const submitForm = ({ appletPassword }: EnterAppletPasswordForm) => {
+      if (isApplet) {
+        submitCallback && submitCallback({ appletPassword });
+
+        return;
+      }
+
       const appletEncryption = encryption || getAppletData(appletsFoldersData, appletId).encryption;
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/Applet/Password/EnterAppletPassword/EnterAppletPassword.tsx:38]: It's not clear in which cases should apply encryption logic.

File: src/modules/Builder/features/SaveAndPublish/SaveAndPublish.tsx [hunk 2]
```diff
@@ -48,12 +55,32 @@ export const SaveAndPublish = () => {
...
+      >
+        {t('saveAndPublish')}
+      </StyledButton>
+      {isNewApplet ? (
+        <AppletPasswordPopup
+          onClose={() => setIsPasswordPopupOpened(false)}
+          popupType={AppletPasswordPopupType.Create}
+          popupVisible={isPasswordPopupOpened}
+          submitCallback={sendRequest}
+        />
```
Comment (kdanilenka32) [inline src/modules/Builder/features/SaveAndPublish/SaveAndPublish.tsx:74]: <AppletPasswordPopup
          onClose={() => setIsPasswordPopupOpened(false)}
          popupType={isNewApplet  ? AppletPasswordPopupType.Create : ...}
          popupVisible={isPasswordPopupOpened}
          submitCallback={sendRequest}
        />

File: src/modules/Dashboard/features/Applet/Password/EnterAppletPassword/EnterAppletPassword.tsx (hunk 1 above)
Comment (IhorNesterenko99) [reply src/modules/Dashboard/features/Applet/Password/EnterAppletPassword/EnterAppletPassword.tsx:38]: if the user inputs password to check it if it's corresponds to the password from the backend

File: src/shared/utils/forms.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import i18n from 'i18n';
+
+export const getDictionaryText = (description?: string | Record<string, string>) => {
+  if (!description) return '';
+
+  const { language } = i18n;
+
+  return typeof description === 'object' ? description[language] ?? '' : description ?? '';
```
Comment (kdanilenka32) [inline src/shared/utils/forms.ts:8]: if this way?
  return (typeof description === 'object' ? description[language]  : description) ?? ""

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**